### PR TITLE
Meets #14592: Status displayed in bulk edit do not respect workflow

### PR DESCRIPTION
**Pull request by slawa85**
[``*`#14592`Status displayed in bulk edit do not respect workflow`](https://www.openproject.org/work_packages/14592)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://netfighter-feature-14592-status-displayed-bulk-edit-not-respect.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by slawa85** [comment]
This PR need to me merged only after this one will be merged https://github.com/opf/openproject/pull/1938

File: app/controllers/work_packages/bulk_controller.rb [hunk 1]
```diff
@@ -40,7 +40,18 @@ class WorkPackages::BulkController < ApplicationController
 
   def edit
     @work_packages.sort!
-    @available_statuses = @projects.map{|p|Workflow.available_statuses(p)}.inject{|memo,w|memo & w}
+    begin
+      @available_statuses = []
+      type = Type.find_by_name('Bug')
```
Comment (ulferts) [inline app/controllers/work_packages/bulk_controller.rb:45]: This shouldn't be here

File: app/controllers/work_packages/bulk_controller.rb [hunk 2]
```diff
@@ -40,7 +40,18 @@ class WorkPackages::BulkController < ApplicationController
...
+      @projects.map do |project|
+        current_user_role = current_user.roles_for_project(project)
```
Comment (ulferts) [inline app/controllers/work_packages/bulk_controller.rb:47]: This disregards the possibility of the current user being the author of the work package. I guess it makes more sense to use #new_statuses_allowed_to on each work_package.

**Comment by ulferts** [comment]
@slawa85 I started reviewing the PR in conjunction with https://github.com/opf/openproject/pull/1938 and then noticed the similarities between the two fixes and the similarities between the work_packages/bulk and the work_packages/moves controller.

I guess it makes sense to extract the common functionality between the two into shared code, e.g. a helper or a service, preferably the later. This service would allow all the modifications necessary for altering a bulk of work packages. As we have the new and edit function in the controllers, we will need the service to have a dry_run option so that it performs no saving.

This might be more work now but it will save us a lot of time later we would otherwise spent fixing twice. What's your opinion?

**Comment by slawa85** [comment]
I can do that, but in detriment to the tickets remaining in the sprint. I guess we will have to handle some of them in the next sprint

**Pull request closed without merging**